File: frontend/src/classes/Game.ts
export interface GameData {
  id: string;
  url: string;
  title: string;
  price: number;
  pageName: string;
  scrapedAt: Date;
  imageUrl: string;
}

export class Game implements GameData {
  id: string;
  url: string;
  title: string;
  price: number;
  pageName: string;
  scrapedAt: Date;
  imageUrl: string;

  constructor(
    id: string,
    url: string,
    title: string,
    price: number,
    pageName: string,
    scrapedAt: Date,
    imageUrl: string,
  ) {
    this.id = id;
    this.url = url;
    this.title = title;
    this.price = price;
    this.pageName = pageName;
    this.scrapedAt = scrapedAt;
    this.imageUrl = imageUrl;
  }
}

File: backend/src/main/java/com/yinnohs/gamesprer/games/domain/Game.java
package com.yinnohs.gamesprer.games.domain;


import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Game {
    private String id;
    private String url;
    private String title;
    private BigDecimal price;
    private String pageName;
    private String imageUrl;
    private LocalDateTime scrapedAt;

}
